test/unit/org/apache/cassandra/batchlog/BatchlogEndpointFilterTest.java

```diff
@@ -163,24 +218,813 @@ public void shouldSelectTwoRandomHostsFromSingleRack() throws UnknownHostExcepti
     @Test
     public void shouldSelectOnlyTwoHostsEvenIfLocal() throws UnknownHostException
     {
+        DatabaseDescriptor.setBatchlogEndpointStrategy(Config.BatchlogEndpointStrategy.random_remote);
         Multimap<String, InetAddressAndPort> endpoints = ImmutableMultimap.<String, InetAddressAndPort> builder()
                                                          .put(LOCAL, InetAddressAndPort.getByName("1"))
                                                          .put(LOCAL, InetAddressAndPort.getByName("11"))
                                                          .build();
-        Collection<InetAddressAndPort> result = filterBatchlogEndpoints(false, endpoints);
+        Collection<InetAddressAndPort> result = filterBatchlogEndpointsRandom(false, endpoints);
         assertThat(result.size()).isEqualTo(2);
         assertTrue(result.contains(InetAddressAndPort.getByName("1")));
         assertTrue(result.contains(InetAddressAndPort.getByName("11")));
     }
 
-    private Collection<InetAddressAndPort> filterBatchlogEndpoints(boolean preferLocalRack, Multimap<String, InetAddressAndPort> endpoints)
+    private Collection<InetAddressAndPort> filterBatchlogEndpointsRandom(boolean preferLocalRack, Multimap<String, InetAddressAndPort> endpoints)
```

Comment:
I think we should rename this function to something like: `filterBatchlogEndpointsRandomFortests` otherwise, when you read the tests you think the original one is called.